src/libertem/common/buffers.py

```diff
@@ -439,6 +447,9 @@ def get_view_for_tile(self, partition, tile):
                 offset = partition_slice.origin[0]
             result_start = tile_slice.origin[0] - offset
             result_stop = result_start + tile_slice.shape[0]
+            # Defend against #1026
+            assert result_start < len(self._data)
+            assert result_stop <= len(self._data)
```

Comment:
Again, maybe explicit `raise`s instead, so they keep working even in "optimized" settings?